api/victoriametrics/v1beta1/vmauth_types.go

```diff
@@ -220,20 +221,87 @@ type VMAuthSpec struct {
+	// SrcPaths is an optional list of regular expressions, which must match the request path.
+	SrcPaths []string `json:"src_paths,omitempty"`
+
+	// SrcHosts is an optional list of regular expressions, which must match the request hostname.
+	SrcHosts []string `json:"src_hosts,omitempty"`
+
+	// UrlPrefix contains backend url prefixes for the proxied request url.
+	URLPrefix []string `json:"url_prefix,omitempty"`
+
+	URLMapCommon `json:",omitempty"`
+}
+
+// URLMapCommon contains common fields for unauthorized user and user in vmuser
+type URLMapCommon struct {
+	// SrcQueryArgs is an optional list of query args, which must match request URL query args.
+	SrcQueryArgs []string `json:"src_query_args,omitempty"`
+
+	// SrcHeaders is an optional list of headers, which must match request headers.
+	SrcHeaders []string `json:"src_headers,omitempty"`
+
+	// DiscoverBackendIPs instructs discovering URLPrefix backend IPs via DNS.
+	DiscoverBackendIPs *bool `json:"discover_backend_ips,omitempty"`
+
+	// RequestHeaders represent additional http headers, that vmauth uses
+	// in form of ["header_key: header_value"]
+	// multiple values for header key:
+	// ["header_key: value1,value2"]
+	// it's available since 1.68.0 version of vmauth
+	// +optional
+	RequestHeaders []string `json:"headers,omitempty"`
+	// ResponseHeaders represent additional http headers, that vmauth adds for request response
+	// in form of ["header_key: header_value"]
+	// multiple values for header key:
+	// ["header_key: value1,value2"]
+	// it's available since 1.93.0 version of vmauth
+	// +optional
+	ResponseHeaders []string `json:"response_headers,omitempty"`
+
+	// RetryStatusCodes defines http status codes in numeric format for request retries
+	// Can be defined per target or at VMUser.spec level
+	// e.g. [429,503]
+	// +optional
+	RetryStatusCodes []int `json:"retry_status_codes,omitempty"`
+
+	// LoadBalancingPolicy defines load balancing policy to use for backend urls.
+	// Supported policies: least_loaded, first_available.
+	// See https://docs.victoriametrics.com/vmauth.html#load-balancing for more details (default "least_loaded")
+	// +optional
+	// +kubebuilder:validation:Enum=least_loaded;first_available
+	LoadBalancingPolicy *string `json:"load_balancing_policy,omitempty"`
+
+	// DropSrcPathPrefixParts is the number of `/`-delimited request path prefix parts to drop before proxying the request to backend.
+	// See https://docs.victoriametrics.com/vmauth.html#dropping-request-path-prefix for more details.
 	// +optional
-	Paths []string `json:"src_paths,omitempty"`
-	// URLs defines url_prefix for dst routing
+	DropSrcPathPrefixParts *int `json:"drop_src_path_prefix_parts,omitempty"`
+}
+
+type UserConfigOption struct {
+	// DefaultURLs backend url for non-matching paths filter
+	// usually used for default backend with error message
+	DefaultURLs []string `json:"default_url,omitempty"`
+
+	TLSCAFile     string `json:"tls_ca_file,omitempty"`
```

Comment:
For now, I suggest to remove tls related flags and implement it as a separate PR later.

 It must support secret references as scrape components do.

Reply:
@f41gh7 please check https://github.com/VictoriaMetrics/operator/pull/951/commits/d8eb187245c5d853f53b0b3bed6cdcb8d2b69c56